### DISCUSSION TITLE
Fix register validation

### DISCUSSION THREAD

**Discussion by maxonfjvipon** [spatie/laravel-multitenancy#290] [Q&A] [answered]
Hello!
I have a problem with register users for tenant.
So I have the following database structure:
1. Landlord
1.1. Tenants
2. Tenant1
2.1. Users

I go to "tenant1.test.test", ```tenant1``` now is current now. Then I try to register new ```user``` for ```tenant1```.
In my controller I have this RegisterRequest
```
<?php

namespace App\Http\Requests\Tenant;

use Illuminate\Foundation\Http\FormRequest;
use Illuminate\Validation\Rules;
use Spatie\Multitenancy\Models\Tenant;

class RegisterRequest extends FormRequest
{
    /**
     * Determine if the user is authorized to make this request.
     *
     * @return bool
     */
    public function authorize(): bool
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array
     */
    public function rules(): array
    {
        return [
            'name' => ['required', 'string', 'max:255'],
            'email' => ['required', 'email', 'string', 'max:255', 'unique:users,email'],
            'password' => ['required', 'confirmed', Rules\Password::defaults()]
        ];
    }
}
```
And validation ```unique``` does not pass. Got this error.
<img width="1118" alt="image" src="https://user-images.githubusercontent.com/37025995/133854855-c7132edd-12a7-4371-b671-5bbada51e70f.png">
it looks for ```users``` table in ```landlord``` database.
If in this request I make dd(...)
```
/**
     * Get the validation rules that apply to the request.
     *
     * @return array
     */
    public function rules(): array
    {
        dd(app('currentTenant'), app('db'));
        return [
            'name' => ['required', 'string', 'max:255'],
            'email' => ['required', 'email', 'string', 'max:255', 'unique:users,email'],
            'password' => ['required', 'confirmed', Rules\Password::defaults()]
        ];
    }
```
Then I can see extension with my ```tenant1``` database
<img width="705" alt="image" src="https://user-images.githubusercontent.com/37025995/133855189-fb292f66-538c-465b-9e63-04a2c8b60e06.png">

So, how I should validate this request for current tenant? Thanks

**Comment by maxonfjvipon** [comment, marked as answer]
Lol, found a solution in 5 minutes. Just added connection name to database, so rules now looks like:
```
 public function rules(): array
    {
        return [
            'name' => ['required', 'string', 'max:255'],
            'email' => ['required', 'email', 'string', 'max:255', 'unique:tenant.users,email'],
            'password' => ['required', 'confirmed', Rules\Password::defaults()]
        ];
    }
``` 
You may also check this one https://github.com/spatie/laravel-multitenancy/issues/154#issuecomment-716602401

**Comment by masterix21** [comment]
Thanks for your share. Happy coding :)
